bazel/grpc_python_deps.bzl

```diff
@@ -49,8 +49,10 @@ def grpc_python_deps():
     if "io_bazel_rules_python" not in native.existing_rules():
         http_archive(
             name = "io_bazel_rules_python",
-            url = "https://github.com/bazelbuild/rules_python/releases/download/0.0.1/rules_python-0.0.1.tar.gz",
-            sha256 = "aa96a691d3a8177f3215b14b0edc9641787abaaa30363a080165d06ab65e1161",
+            url = "https://github.com/bazelbuild/rules_python/releases/download/0.4.0/rules_python-0.4.0.tar.gz",
+            sha256 = "954aa89b491be4a083304a2cb838019c8b8c3720a7abb9c4cb81ac7a24230cea",
+            patches = ["//third_party:rules_python.patch"],
```

Comment:
Does this break users of the gRPC library?

I'm trying to upgrade a repository that uses gRPC to the latest gRPC release (1.44.0). This line seems to be causing an error (repo details omitted) when building with bazel 4.2.2:

```
bazel test ${MY_GRPC_TEST_TARGET}
INFO: Repository io_bazel_rules_python instantiated at:  
  ${MY_REPO}/WORKSPACE.bazel:26:5: in <toplevel>  
  ${MY_REPO}/bazel/deps.bzl:59:14: in deps.
 /home/alexmc/.cache/bazel/_bazel_alexmc/e5e83162f61030880de4b3e9d73ac179/external/com_github_grpc_grpc/bazel/grpc_deps.bzl:459:21: in grpc_deps
 /home/alexmc/.cache/bazel/_bazel_alexmc/e5e83162f61030880de4b3e9d73ac179/external/com_github_grpc_grpc/bazel/grpc_python_deps.bzl:53:21: in grpc_python_deps
Repository rule http_archive defined at:  
 /home/alexmc/.cache/bazel/_bazel_alexmc/e5e83162f61030880de4b3e9d73ac179/external/bazel_tools/tools/build_defs/repo/http.bzl:336:31: in <toplevel>

INFO: Repository 'io_bazel_rules_python' used the following cache hits instead of downloading the corresponding file.
* Hash '954aa89b491be4a083304a2cb838019c8b8c3720a7abb9c4cb81ac7a24230cea' for https://github.com/bazelbuild/rules_python/releases/download/0.4.0/rules_python-0.4.0.tar.gz
If the definition of 'io_bazel_rules_python' was updated, verify that the hashes were also updated.
ERROR: An error occurred during the fetch of repository 'io_bazel_rules_python':
   Traceback (most recent call last):
        File "/home/alexmc/.cache/bazel/_bazel_alexmc/e5e83162f61030880de4b3e9d73ac179/external/bazel_tools/tools/build_defs/repo/http.bzl", line 121, column 10, in _http_archive_impl
                patch(ctx)
        File "/home/alexmc/.cache/bazel/_bazel_alexmc/e5e83162f61030880de4b3e9d73ac179/external/bazel_tools/tools/build_defs/repo/utils.bzl", line 135, column 22, in patch
                ctx.patch(patchfile, strip)
Error in patch: Unable to load package for //third_party:rules_python.patch: BUILD file not found in any of the following directories. Add a BUILD file to a directory to mark it as a package.
 - ${MY_REPO}/third_party
```

Specifically, it's looking for a third_party directory to exist as a sibling directory to the directory containing the grpc source archive.

Is this expected? I'm wondering if this dependency on a `//target_path:target_name` doesn't work when this code is included from an external repository.

Reply:
Thanks for the report!

We have [a distribtest](https://github.com/grpc/grpc/tree/master/bazel/test/python_test_repo) that should catch things like this and -- in fact --  I'm not able to reproduce your build error with it. Adding the repository to the path here should be harmless, so I've put together a PR to do that.

I'd like to try to figure out why our distribtest isn't reproducing this so we can catch this sort of thing in the future. I'd be grateful if you could help put together a minimal reproduction repo.

Reply:
Here's a minimal repro: https://github.com/CodingCanuck/grpc-build-error-1-42-rules_python.patch

It looks like the build error happens when trying to load anything from `@io_bazel_rules_python` (or perhaps `@io_bazel_rules_python//python`) *after* calling `grpc_deps()`.

Reply:
Ah, I see. Thanks for following up. I'll add a case like this to our disribtests.